src/BSON/ObjectId.h

```diff
@@ -21,6 +21,6 @@
 
 #include <php.h>
 
-void phongo_objectid_init(zval* return_value, const bson_oid_t* oid);
+void php_phongo_bson_new_objectid(zval* return_value, const bson_oid_t* oid);
```

Comment:
I've refactored the factories to return a bool and updated all instances to handle the return value. The only place I wasn't able to reliably handle were the compatibility macros for int64, as the macro itself can't decide what to do in case construction of the Int64 instance goes wrong. Since the factory currently always returns true, I'm ignoring that for the time being.

Reply:
> Since the factory currently always returns true, I'm ignoring that for the time being.

Please leave a comment in or around the macro if you haven't already.